src/Build/BackEnd/Shared/TargetResult.cs

```diff
@@ -141,6 +146,18 @@ internal bool TargetFailureDoesntCauseBuildFailure
             set => _targetFailureDoesntCauseBuildFailure = value;
         }
 
+        /// <summary>
+        /// Sets or gets a flag indicating whether at least one target which has run after us (transitively via AfterTargets) failed.
+        /// </summary>
+        internal bool AfterTargetsHaveFailed
```

Comment:
why is a field needed behind this property?  `internal bool AfterTargetsHaveFailed { get; set; }`

Reply:
I see: a field is needed, as the value is passed by ref later in the code